plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1407 @@
+
+        if config.get('view'):
+            views_path = "data/ietf-snmp:snmp/vacm"
+            payload = self.build_create_view_payload(config)
+            views_request = {'path': views_path, 'method': method, 'data': payload}
+            requests.append(views_request)
+
+        return requests
+
+    def get_agentaddress_requests(self, config, have, overridden_or_replaced):
+        """ Build the payload for SNMP agentaddress
+        :rtype: A dictionary
+        :returns: The payload for SNMP agentaddress
+        """
+        agentaddress = config.get('agentaddress')
+        have_agentaddress = have.get('agentaddress')
+        have_agentaddress_names = []
+        if have_agentaddress and not overridden_or_replaced:
+            have_agentaddress_names = [agent_entry.get('name') for agent_entry in have_agentaddress]
+        agentaddress_list = []
+        agentaddressdict = {}
+        payload_url = {}
+        for conf in agentaddress:
+            name = ""
+            agentaddress_dict = {}
+            if 'name' in conf:
+                matching_name = next((agent_entry for agent_entry in have_agentaddress if agent_entry.get('name') == conf.get('name')), None)
+                matching_vrf = matching_name.get('vrf') == conf.get('vrf')
+                matching_interface = matching_name.get('interface') == conf.get('interface')
+                matching_ip = matching_name.get('ip') == conf.get('ip')
+                if matching_name and (matching_vrf or matching_ip or matching_interface):
+                    self._module.fail_json(msg="There already exists an agentaddress with that name")
+                else:
+                    name = conf.get('name')
+    
+            vrf = ""
+            interface = conf.get('interface')
+            # if there is already a matching 'ip' and 'vrf' then use that one
+            if have_agentaddress and 'ip' in conf and ('vrf' in conf or 'interface' in conf):
+                vrf = conf.get('vrf')
+                if vrf:
+                    interface = vrf
+                    matching_agentaddress = next((agent_entry for agent_entry in have_agentaddress
+                                                  if agent_entry.get('ip') == conf.get('ip')
+                                                  and (agent_entry.get('interface') == conf.get('vrf')
+                                                       or agent_entry.get('interface') == conf.get('interface'))),
+                                                 None)
+                    if 'name' not in conf and matching_agentaddress:
+                        name = matching_agentaddress.get('name')
+            if len(name) == 0:
+                name = self.get_agententry(have_agentaddress_names)
+
+            agentaddress_dict['name'] = name
+            agentaddress_dict['udp'] = {'ietf-snmp-ext:interface': interface, 'ip': conf.get('ip'), 'port': conf.get('port')}
```

Comment:
Neither interface nor port is required. If they aren't specified, don't specify them.

```suggestion
            agentaddress_dict['udp'] = {'ip': conf.get('ip')}
            if interface:
                agentaddress_dict['udp']['ietf-snmp-ext:interface'] = interface
            if conf.get('port')
                agentaddress_dict['udp'][port'] = conf['port']
```